integrated_channels/canvas/client.py

```diff
@@ -295,20 +353,38 @@ def _get_course_id_from_integration_id(self, integration_id):
                 all_courses_response.status_code
             )
 
-        course_id = None
+        course_found = None
         for course in all_courses_response:
-            if course['integration_id'] == integration_id:
-                course_id = course['id']
+            if course['integration_id'] == edx_course_id:
+                course_found = course
                 break
+        return course_found
 
-        if not course_id:
+        # now let's try under all subaccounts
```

Comment:
Need to do this in case course was posted under a different subaccount / account previously

Reply:
I did not implement this part yet because I want to hear from reviewers what they think of this. Basically doing this means that we won't possibly be able to support multiple canvas accounts per customer., because we would be reaching out to find courses under all other subaccounts too. Meaning effecitvely merging in all subaccounts as if. they are all one set of courses (or ignoring the subaccounts in some way).

Should we only support one subaccount per customer? Should we try to find courses under other subaccounts?

With customers we don't yet have an issue since we only use one subaccount per customer so far. But our own demo instance uses multiple subaccounts. If we assume a course can only be posted under one subaccount, then we don't need to do this and we can simply acknowledge a failure to create a course in that case (after checking response of course)

Reply:
Also, FYI it's easy to move course between subaccounts in Canvas (manually or via API), if needed

Reply:
In with the convo going on with Adrienne's [comment](https://github.com/edx/edx-enterprise/pull/1279/files#r657913079), I'm wondering if we are able to hit the generic `GET /api/v1/courses` and get all courses in all sub-accounts. I think we should test out the functionality because it could simplify things.